mlrun/projects/pipelines.py

```diff
@@ -706,56 +751,70 @@ def run(
         secrets=None,
         artifact_path=None,
         namespace=None,
+        source=None,
     ) -> typing.Optional[_PipelineRunStatus]:
         workflow_name = name.split("-")[-1] if f"{project.name}-" in name else name
-        runner_name = f"workflow-runner-{workflow_name}"
+
         run_id = None
 
-        if workflow_spec.schedule and not project.spec.is_remote():
+        # If the user provided a source we want to load the project from the source
+        # (like from a specific commit/branch from git repo) without changing the source of the project (save=False).
+        save, current_source = (
+            (False, source) if source else (True, project.spec.source)
+        )
+        if "://" not in current_source:
```

Comment:
This if statement will yield and error if `current_source` is `None`:

```
TypeError: argument of type 'NoneType' is not iterable
```

I see default value for source is usually None or I missed something.

Reply:
if the source is `None` then `current_source=project.spec.source` which is by default `""`, you can see it in the line above